java/vector/src/main/java/org/apache/arrow/vector/Float2Vector.java

```diff
@@ -0,0 +1,434 @@
+   * @param allocator allocator for memory management.
+   */
+  public Float2Vector(String name, FieldType fieldType, BufferAllocator allocator) {
+    this(new Field(name, fieldType, null), allocator);
+  }
+
+  /**
+   * Instantiate a Float2Vector. This doesn't allocate any memory for
+   * the data in vector.
+   *
+   * @param field field materialized by this vector
+   * @param allocator allocator for memory management.
+   */
+  public Float2Vector(Field field, BufferAllocator allocator) {
+    super(field, allocator, TYPE_WIDTH);
+  }
+
+  @Override
+  protected FieldReader getReaderImpl() {
+    return new Float2ReaderImpl(Float2Vector.this);
+  }
+
+  /**
+   * Get minor type for this vector. The vector holds values belonging
+   * to a particular type.
+   *
+   * @return {@link MinorType}
+   */
+  @Override
+  public MinorType getMinorType() {
+    return MinorType.FLOAT2;
+  }
+
+
+  /*----------------------------------------------------------------*
+   |                                                                |
+   |          vector value retrieval methods                        |
+   |                                                                |
+   *----------------------------------------------------------------*/
+
+
+  /**
+   * Get the element at the given index from the vector.
+   *
+   * @param index   position of element
+   * @return element at given index
+   */
+  public short get(int index) throws IllegalStateException {
+    if (NULL_CHECKING_ENABLED && isSet(index) == 0) {
+      throw new IllegalStateException("Value at index is null");
+    }
+    return valueBuffer.getShort((long) index * TYPE_WIDTH);
+  }
+
+  /**
+   * Get the element at the given index from the vector and
+   * sets the state in holder. If element at given index
+   * is null, holder.isSet will be zero.
+   *
+   * @param index   position of element
+   */
+  public void get(int index, NullableFloat2Holder holder) {
+    if (isSet(index) == 0) {
+      holder.isSet = 0;
+      return;
+    }
+    holder.isSet = 1;
+    holder.value = valueBuffer.getShort((long) index * TYPE_WIDTH);
+  }
+
+  /**
+   * Same as {@link #get(int)}.
+   *
+   * @param index   position of element
+   * @return element at given index
+   */
+  @Override
+  public Short getObject(int index) {
+    if (isSet(index) == 0) {
+      return null;
+    } else {
+      return valueBuffer.getShort((long) index * TYPE_WIDTH);
+    }
+  }
+
+  /**
+   * Given a data buffer, get the value stored at a particular position
+   * in the vector.
+   *
+   * <p>This method should not be used externally.
+   *
+   * @param buffer data buffer
+   * @param index position of the element.
+   * @return value stored at the index.
+   */
+  static short get(final ArrowBuf buffer, final int index) {
+    return buffer.getShort((long) index * TYPE_WIDTH);
+  }
+
+  @Override
+  public double getValueAsDouble(int index) {
+    return this.get(index);
```

Comment:
So this converts the short into a double directly? Isn't this wrong?

Reply:
I am goin to change to:
````
  @Override
  public double getValueAsDouble(int index) {
    return getValueAsFloat(index);
  }
````
The process of Widening Casting (automatically) involves converting a smaller type (float) size to a larger type size (double). A problem arises if the user needs to compare the initial and final values for Numbers expressed in floating point.

@lidavidm Should this method be maintained or should an exception be thrown?

Reply:
Why would we throw an exception?

Reply:
For rounding like this for example:

````
Input:
float f32 = 1.618034F;
short f16short = Float16.toFloat16(f32);
System.out.println("Float 32: " + f32);
System.out.println("Short: " + f16short);
System.out.println("Float 16: " + Float16.toFloat(f16short));
System.out.println("float16 to double: " + (double) Float16.toFloat(f16short));

Ouput:
Float 32: 1.618034
Short: 15993
Float 16: 1.6181641
float16 to double: 1.6181640625
````

Reply:
The point of this method is that we're doing the conversion for the user (and they should be aware of potential issues), right?

Reply:
It is clear now, keep them as they are.

Reply:
The implementation is still wrong, though. We should never let short implicitly cast to something else in this class.

Reply:
That was my fault, I just changed it